### PR TITLE
Hosted/Ftdi

### PR DESCRIPTION
**Pull request by UweBonnes**
Fix a crash introduced  with 7b1eb6e6e3 and another visible with ftdi/mpsse/SWD

### CODE REVIEW

**Review by DX-MON — CHANGES_REQUESTED** [review]
Looks good, there are a couple of changes requested but we see no reason once those are looked at and dealt with not to merge this.

File: src/platforms/hosted/bmp_libusb.c [hunk 1]
```diff
@@ -146,20 +146,19 @@ int find_debuggers(BMP_CL_OPTIONS_t *cl_opts,bmp_info_t *info)
 	char serial[64];
 	char manufacturer[128];
 	char product[128];
-	bmp_type_t type;
 	bool access_problems = false;
-	char *active_cable = NULL;
```
[the remark this answers is not in the document]
Comment (UweBonnes) [reply src/platforms/hosted/bmp_libusb.c:151]: Removed.

File: src/platforms/hosted/bmp_libusb.c [hunk 2]
```diff
@@ -299,8 +298,9 @@ int find_debuggers(BMP_CL_OPTIONS_t *cl_opts,bmp_info_t *info)
 	}
 	if ((found_debuggers == 0) && ftdi_unknown)
 		DEBUG_WARN("Generic FTDI MPSSE VID/PID found. Please specify exact type with \"-c <cable>\" !\n");
-	if ((found_debuggers == 1) && !cl_opts->opt_cable && (type == BMP_TYPE_LIBFTDI))
-		cl_opts->opt_cable = active_cable;
+	if ((found_debuggers == 1) && !cl_opts->opt_cable &&
```
[the remark this answers is not in the document]
Comment (UweBonnes) [reply src/platforms/hosted/bmp_libusb.c]: Changed

**Review by DX-MON — APPROVED** [review]
Looks good, merging.

**Pull request merged**